### PR TITLE
Smart recording 24.0.0

### PR DESCRIPTION
**Pull request by eddyStreamlabs**

### CODE REVIEW

File: libobs/obs-video.c [hunk 1]
```diff
@@ -664,22 +710,62 @@ static inline void copy_rgbx_frame(struct video_frame *output,
...
+		locked = video_output_lock_frame(video->video,
+						 &main_output_frame, count,
+						 main_input_frame->timestamp,
+						 OBS_MAIN_VIDEO_RENDERING);
+	} else {
+		locked = video_output_lock_frame(
+			video->video, &streaming_output_frame, count,
+			streaming_input_frame->timestamp,
+			OBS_STREAMING_VIDEO_RENDERING);
+		locked = video_output_lock_frame(
```
Comment (summeroff) [inline libobs/obs-video.c]: can be **locked &= video_output_lock_frame(** to have kind of correct value in locked flag for both outputs.

Comment (eddyStreamlabs) [reply libobs/obs-video.c]: Good catch, I actually fixed it in another sub branch I have, forgot to put it in here.

File: libobs/obs-video.c [hunk 2]
```diff
@@ -728,27 +814,52 @@ static inline void output_frame(bool raw_active, const bool gpu_active)
...
 	int prev_texture = cur_texture == 0 ? NUM_TEXTURES - 1
 					    : cur_texture - 1;
-	struct video_data frame;
+	struct video_data main_frame;
+	struct video_data streaming_frame;
+	struct video_data recording_frame;
 	bool frame_ready = 0;
 
-	memset(&frame, 0, sizeof(struct video_data));
+	memset(&main_frame, 0, sizeof(struct video_data));
```
Comment (summeroff) [inline libobs/obs-video.c]: memset can be replaced with struct initialization to zero like  **= {0}** ?

Comment (eddyStreamlabs) [reply libobs/obs-video.c]: Done 👍

**Pull request merged**